### ISSUE TITLE
[Bug]: Issue with Adding Custom Cakes

### ISSUE THREAD

**Issue by SaihaoDraw** [Mrbysco/TelePastries#22]
### General Info

- [X] I am running a modpack
- [ ] I can reproduce this issue consistently in single-player
- [ ] I can reproduce this issue consistently in multi-player
- [X] I have searched for this issue previously and it was either (1) not previously reported, or (2) previously fixed and I am having the same problem.
- [ ] I am crashing and can provide my crash report(s)
- [ ] I am using the latest version of the mod

### Forge version

forge 47.3.5

### Minecraft version

1.20.1

### Mod version

1.20.1.0.8.3

### Java version

8

### Issue Description

Hello,

I've decided to continue the conversation here on GitHub as it's the most appropriate place.

I'm encountering an issue with adding custom cakes. I modified the base cakes and tried to add a new one, but every time I restart the game, the custom cakes are removed from the config file.

I've double-checked the syntax multiple times. I'm using the same structure as for CustomCake3 and only changed the relevant numbers in the config. Despite this, I receive the following error in my logs:

Incorrect key CustomCake4 was corrected from SimpleCommentedConfig:{consumeCustomCake4=true, customCake4Name=Bumble Cake, customCake4Dimension=the_bumblezone:the_bumblezone, customCake4RefillItem=[the_bumblezone:beespire_catalyst], customCake4MinY=2, customCake4MaxY=254} to its default, null.

I'm a bit lost at this point 😅. Any help would be appreciated!

**Comment by Mrbysco** [comment]
Are you removing line-breaks? as you can't touch those

**Comment by SaihaoDraw** [comment]
No, normally not.

The basic custom cakes (1, 2, 3) all work perfectly. But when I add the 4 and then launch the game. It is automatically deleted when the game is loaded.

![Capture](https://github.com/user-attachments/assets/d0c31b1d-27fd-45e5-8603-d35c78af8281)

**Comment by Mrbysco** [comment]
Wait, is the refill item an array in that screenshot? I think it should just be 
`customCake3RefillITem = "blue_skies:everdawn_catalyst"` etc

**Comment by SaihaoDraw** [comment]
Yes, but it's the same for custom cake 1 and 2. I just copied and pasted and changed the number to 4. The first three custom cakes also work with arrays, and everything functions properly

![Capture](https://github.com/user-attachments/assets/7043ecf1-dc08-490a-9a1b-672f32237710)

**Comment by Mrbysco** [comment]
I'll look into it this weekend

**Comment by Mrbysco** [comment]
Have you tried editing it in-game using a mod like Configured btw?

**Comment by SaihaoDraw** [comment]
I just tried adding custom cakes via the in-game config, but it's impossible to add new custom cakes that way. I can modify existing ones slightly, but adding anything new doesn’t work.

**Comment by Mrbysco** [comment]
There is no way to add more than the existing ones currently

**Comment by SaihaoDraw** [comment]
Do you think it would be possible to add an option or a way to create new custom cakes in a future update ?

**Comment by Mrbysco** [comment]
I wouldn't know what the best way would be. I think you should be able to add new ones using KubeJs

**Comment by SaihaoDraw** [comment]
I just looked into making a custom cake with KubeJS. After asking on Discord, I was advised to use the Farmer's Delight extension to create the custom pies. Additionally, I need to add a teleportation code. There's also the risk of spawning and suffocating in blocks.

Overall, it's complex and quite a hassle xD.

I hope it will be possible for you to add an option or something similar to make it easier to add custom cakes.

Thank you for your help! I'll keep an eye out for any updates!
